src/settings/SettingsAddress.js

```diff
@@ -58,18 +57,7 @@ export default class SettingsAddress extends React.Component {
 			address_postcode: state.postcode,
 			phone: state.tel,
 		};
-		LiveData.api.setAccountSettings(req).then(
-			response => {
-				if (response.set_settings === 1) {
-					this.actions.updateSettingFields(req);
-				} else {
-					SettingsAddress.handleUpdateError(response);
-				}
-			},
-			response => {
-				SettingsAddress.handleUpdateError(response);
-			}
-		);
+		this.props.actions.attemptUpdateSettings(req, SettingsAddress.handleUpdateError);
```

Comment:
Yea, actions should not return data, at least from it's name it should change state, but not returning any of them.

So are you suggesting we keep the api call in components?